src/lib/api/pricer/getCurvePrice.ts

```diff
@@ -43,8 +48,22 @@ export async function getCurvePrice({
     `https://api.curve.fi/api/getPools/${chainPrefix}/main`
   )
   const parsed = respSchema.parse(resp.data)
-  const filterCoin = parsed.data.poolData
-    .flatMap((pool) => pool.coins)
-    .filter((coin) => coin.address === asset)
-  return filterCoin?.[0]?.usdPrice
+  const poolAsset = parsed.data.poolData.filter(
+    (pool) => pool.lpTokenAddress === asset
+  )
+
+  let lpTokenPrice = 0
+  poolAsset.map((pool) => {
```

Comment:
This isn't really how map is supposed to be used. The purpose of map is to return a new array based on some transformation of each item in the original array. You should use reduce for this or forEach. Reduce is a little intense because you're reducing on two levels but it would look something like this:

```ts
const lpTokenPrice = poolAsset.reduce((summed, pool) => (
  summed + pool.coins.reduce((summedUnderlying, coin) => (
    summedUnderlying + coin.usdPrice * parseFloat(...)
  ), 0)
), 0) 
```

Reply:
That's not really accurate as you're doing more maths here (not just summing) but you get the idea I think.

Reply:
You're right about map.